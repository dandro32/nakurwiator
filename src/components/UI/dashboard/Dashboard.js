import React from 'react';
import Card from '../card/Card.js';
import PropTypes from 'prop-types';

const Dashboard = ({cards}) => {
  const dashboardCards = cards.map((card, i) => {
    return <Card key={i} card={card} />;
  });

  return (
    <div className="row dashboard-containter justify-content-around align-items-center">
      {dashboardCards}
    </div>
  );
};

Dashboard.propTypes = {
  cards: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      desc: PropTypes.string.isRequired,
      href: PropTypes.string.isRequired,
      hrefTitle: PropTypes.string.isRequired,
      imgSrc: PropTypes.string.isRequired,
    })
  )
};

export default Dashboard;