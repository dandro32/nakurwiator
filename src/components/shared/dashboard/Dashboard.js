import React from 'react';
import Card from '../card/Card.js';
import PropTypes from 'prop-types';

const Dashboard = ({cards}) => {
  const dashboardCards = cards.map((card, i) => {
    return <Card key={i} card={card} />;
  });

  return (
    <div className="row d-flex justify-content-around align-items-center">
      {dashboardCards}
    </div>
  );
};

Dashboard.propTypes = {
  cards: PropTypes.array.isRequired
};

export default Dashboard;