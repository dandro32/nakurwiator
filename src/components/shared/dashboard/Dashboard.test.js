import React from 'react';
import Dashboard from './Dashboard';
import TestRenderer from 'react-test-renderer';

jest.mock('../card/Card', () => 'Card component');

describe('Dashboard component test', () => {
  const mockCarsdData = [
    {test1: 'test'},
    {test2: 'test'},
    {test3: 'test'}
  ];
  let renderedDashboard;
    
  beforeEach(() => {
    renderedDashboard = TestRenderer.create(
      <Dashboard cards={mockCarsdData}/>
    ).toJSON();
  });

  it('should render dashboard correctly', ()=> {
    expect(renderedDashboard).toMatchSnapshot();
  });

  it('should render dashboard cards exact count', ()=> {
    expect(renderedDashboard.children.length).toBe(3);
  });


});