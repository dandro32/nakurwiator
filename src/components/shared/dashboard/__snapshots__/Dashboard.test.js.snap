// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component test should render dashboard correctly 1`] = `
<div
  className="row justify-content-center align-self-center"
>
  <Card component
    card={
      Object {
        "test1": "test",
      }
    }
  />
  <Card component
    card={
      Object {
        "test2": "test",
      }
    }
  />
  <Card component
    card={
      Object {
        "test3": "test",
      }
    }
  />
</div>
`;
